src/sage/rings/finite_rings/element_pari_ffelt.pyx

```diff
@@ -496,11 +498,8 @@ cdef class FiniteFieldElement_pari_ffelt(FinitePolyExtElement):
             self.construct_from(x.constant_coefficient())
 
         elif isinstance(x, list):
-            if len(x) == self._parent.degree():
-                self.construct_from(self._parent.vector_space(map=False)(x))
-            else:
-                Fp = self._parent.base_ring()
-                self.construct_from(self._parent.polynomial_ring()([Fp(y) for y in x]))
+            Fp = self._parent.base_ring()
+            self.construct_from(self._parent.polynomial_ring()([Fp(y) for y in x]))
```

Comment:
Can the poly ring take a `map(Fp, x)` as input? This might be faster by avoiding a transient list (this should be checked in isolation).

Reply:
The case where the caller passes a large list (longer than the field degree) seems very uncommon.
The polynomial constructor recognizes `list` and `tuple` but not iterators.